Mlem/Models/Trackers/Comment Tracker.swift

```diff
@@ -11,4 +11,16 @@ class CommentTracker: ObservableObject
 {
     @Published var comments: [HierarchicalComment] = .init()
     @Published var isLoading: Bool = true
+    
+    /// A method to add new comments into the tracker, duplicate comments will be rejected
+    func add(_ newComments: [HierarchicalComment]) {
```

Comment:
We're iterating inside an iteration here which will scale badly, we should look to implement this function similarly to the approach in the post tracker using a backing set:

```swift
    /// A method to add new posts into the tracker, duplicate posts will be rejected
    /// - Parameter newPosts: The array of `APIPostView` you wish to add
    func add(_ newPosts: [APIPostView]) {
        let accepted = newPosts.filter { ids.insert($0.id).inserted }
        posts.append(contentsOf: accepted)
    }
```

Reply:
Definitely true.  The whole idea of hierarchical comments doesn't make a _ton_ of sense for profile view, but does if you click on one.